### PR TITLE
Add is none function

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
Add support `is_none` in the condition

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
```python
import random
import typing
from flytekit import conditional, task, workflow


@task
def coin_toss(seed: int) -> typing.Optional[bool]:
    r = random.Random(seed)
    if r.random() < 0.5:
        return True
    return None


@task
def failed() -> int:
    return -1


@task
def success() -> int:
    return 0


@workflow
def wf(seed: int = 5) -> int:
    result = coin_toss(seed=seed)
    return conditional("test").if_(result.is_none()).then(success()).else_().then(failed())


if __name__ == '__main__':
    wf()
```

image

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3514

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/core/promise.py [hunk 1]
```diff
@@ -167,6 +181,8 @@ def eval(self) -> bool:
 
         if isinstance(self.rhs, Promise):
             rhs = self.rhs.eval()
+        elif self.rhs.scalar.none_type:
```
Comment (wild-endeavor) [inline flytekit/core/promise.py:184]: why is this only added for rhs?  should we also add it for lhs?

Comment (pingsutw) [reply flytekit/core/promise.py:186]: updated it

File: flytekit/models/core/condition.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def from_flyte_idl(cls, pb2_object):
 
 
 class Operand(_common.FlyteIdlEntity):
-    def __init__(self, primitive=None, var=None):
+    def __init__(self, primitive=None, var=None, scalar=None):
         """
         Defines an operand to a comparison expression.
         :param flytekit.models.literals.Primitive primitive:
```
Comment (wild-endeavor) [inline flytekit/models/core/condition.py:140]: can you add scalar to the params list

**Review by wild-endeavor — APPROVED** [review]

File: flytekit/models/core/condition.py (hunk 2 above)
Comment (pingsutw) [reply flytekit/models/core/condition.py]: updated

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**